web/controllers/api/docs_controller.ex

```diff
@@ -163,16 +163,13 @@ defmodule HexWeb.API.DocsController do
     HexWeb.CDN.purge_key(:fastly_hexdocs, unversioned_key)
     HexWeb.CDN.purge_key(:fastly_hexdocs, versioned_key)
 
-    # Set docs flag on release
-    HexWeb.Repo.transaction(fn ->
-      Ecto.Changeset.change(release, has_docs: true)
-      |> HexWeb.Repo.update!
+    multi =
+      Ecto.Multi.new
+      |> Ecto.Multi.update(:release, Ecto.Changeset.change(release, has_docs: true))
+      |> Ecto.Multi.update(:package, Ecto.Changeset.change(release.package, docs_updated_at: Ecto.DateTime.utc))
+      |> Ecto.Multi.run(:log, fn _ -> audit(user, "docs.publish", {package, release}) end)
```

Comment:
Since this audit log doesn't depend on data from previous steps, it could be simply:

`|> Ecto.Multi.insert(:log, audit(user, "docs.publish", {package, release})`

however I made it that `audit/3` calls Repo.insert under the hood. Moreover, I think it would be weird to sometimes log with Multi.insert and sometimes with Multi.run. I think it would be neat to allow passing fun to Ecto.insert as proposed here: https://groups.google.com/forum/#!topic/elixir-ecto/zrJtqiR0-gU